nixos/modules/services/security/aesmd.nix

```diff
@@ -148,8 +252,16 @@ in
           BindReadOnlyPaths = [
             builtins.storeDir
             # Hardcoded path AESM_CONFIG_FILE in psw/ae/aesm_service/source/utils/aesm_config.cpp
-            "${configFile}:/etc/aesmd.conf"
-          ];
+            "${aesmdConfigFile}:/etc/aesmd.conf"
+          ]
+          ++ optional (!isNull cfg.qcnl.settings) (let
+            toSnakeCase = replaceStrings upperChars (map (s: "_${s}") lowerChars);
+            qcnlConfig = builtins.toJSON (mapAttrs' (name: value: nameValuePair (toSnakeCase name) value) (filterAttrs (n: v: !isNull v) cfg.qcnl.settings));
```

Comment:
it's always camelCase -> snake_case conversion and the allowed fields are already explicitly set via options, so this is not user-facing. The only way how this could break, would be if we added an option, which is not camel case and/or Intel would introduce a non-snake-case option

Reply:
Fine for me 👍🏻